src/app/selectors.js

```diff
@@ -188,6 +188,43 @@ const groupModelsByOwner = modelData => {
   return grouped;
 };
 
+/**
+  Returns the number of models, applications and units.
+  @param {Object} countModelData
+  @returns {Function} The set of model data counts.
+*/
+const countModelData = modelData => {
+  let machinesCount = 0;
+  let applicationCount = 0;
+  let unitCount = 0;
+
+  for (const modelUUID in modelData) {
+    const applications = modelData[modelUUID].applications;
+    for (const applicationID in applications) {
+      if (applications.hasOwnProperty(applicationID)) {
+        applicationCount++;
+      }
+      const units = applications[applicationID].units;
```

Comment:
There isn't really any need to loop through the units to get this value. You can instead do the following and remove the loop. You can do the same for the machines below.

```suggestion
      unitCount = Object.keys(applications[applicationID].units).length;
```